src/services/http_request_service.ts

```diff
@@ -28,6 +28,44 @@ export default class HttpRequestService {
   // FILE MARKER - PUBLIC //////////////////////////////////////////////////////
   //////////////////////////////////////////////////////////////////////////////
 
+  /**
+   * @description
+   *     Checks if the incoming request accepts the type(s) in the parameter.
+   *     This method will check if the requests `Accept` header contains
+   *     the passed in types
+   *
+   * @param string|string[] type
+   *     The content-type/mime-type(s) to check if the request accepts it
+   *
+   * @example
+   *     // YourResource.ts - assume the request accepts "text/html"
+   *     const isAccepted = this.request.accepts("text/html"); // "text/html"
+   *     // or can also pass in an array and will match on the first one found
+   *     const isAccepted = this.request.accepts(["text/html", "text/xml"]); // "text/html"
+   *     // and will return false if not found
+   *     const isAccepted = this.request.accepts("text/xml"); // false
+   *
+   * @return boolean|string
+   *     False if the request doesn't accept any of the passed in types,
+   *     or the content type that was matches
+   */
+  public accepts (request: any, type: string|string[]): boolean|string {
+    const acceptHeader = request.headers.get('Accept')
+    const acceptedContentTypes = acceptHeader.split(';')[0];
```

Comment:
As I expressed in https://github.com/drashland/deno-drash/issues/173#issuecomment-613671510, a client might pass multiple things that the function could not handle...
```json
{
    "Accept": "application/json;text/html;"
}
```
```json
{
    "Accept": " application/json; text/html;"
}
```
```json
{
    "Accept": "application/json; text/html"
}
```
Now the results of these dummy jsons:
```typescript
// Example 1
console.log(acceptContentTypes); // Outputs ["application/json","text/html"]

// Example 2
console.log(acceptContentTypes); // Outputs [" application/json"," text/html"]

// Example 3
console.log(acceptContentTypes); // Outputs [" application/json"]
```
A working solution is to remove extra spaces with regex.
```typescript
const acceptedContentTypes = acceptHeader.replace(/\s/g, "").split(';')

// Example 1
console.log(acceptContentTypes); // Outputs ["application/json","text/html",""]

// Example 2
console.log(acceptContentTypes); // Outputs [" application/json","text/html",""]

// Example 3
console.log(acceptContentTypes); // Outputs [" application/json","text/html"]
```

Having the extra element in the array will not affect the rest of the implementation.

Reply:
That implementation shouldn't be necessary right? I get your point actually if the Accept header is `application/json;text/html`, what I implemented would actually ignore the `text/html` part (didn't realise the content types can also be seperated by `;`). But because the `acceptedContentTypes` is just a string, even doing `' something, hello; text/html; application/json'.indexOf(type)` should still be able to capture if the `type` exists? Eg there shouldn't be a need to trim spaces or turn it into an array?

```
const acceptHeader = request.headers.get('Accept'); // "  text/html;0.5; someother stuff"
const split= acceptHeader.split(';'); // ["  text/html", "0.5", "someother stuff"];
const contentTypes = split[0]; // "  text/html"
contentTypes.indexOf("text/html") >= 0; // true
```

Reply:
```typescript
const str = "hello; goodbye";
const array1 = str.split(";"); // ["hello", " goodbye"]
console.log(array1.indexOf("goodbye")); // Outputs -1 (didn't find element)
```
```typescript
const str = "hello; goodbye";
const array1 = str.replace(/\s/g, "").split(";"); // ["hello", "goodbye"]
console.log(array1.indexOf("goodbye")); // Outputs 1 (found element)
```
The reason I'm saying this is because, even though browsers remove all blank spaces in their requests, the client (for example, postman/postwoman) doesn't.

Reply:
Yeah fair enough actually, if content types could be in 1st and 2nd index, I can't do `.split(';')[0]` so i'll have to just split, and remove the spaces then. Nice one!

Reply:
Oh, btw. In the *working solutions* I proposed, I blundered with copy and paste. There are no spaces in the example 2 and 3. Should be `["application/JSON","text/html"]` in both cases.

Reply:
Had to change it up a bit as `str.replace(...).split(';')` might equal `["text/html,text/plain", "...", "..."]` and a `thatArray.indexOf("text/html")` wouldnt work